// react-hero-carousel/dist/HeroCarousel/HeroCarousel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CustomInterval = exports.BackgroundImages = exports.Images = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

require("reset-css");

var _HeroCarousel = _interopRequireDefault(require("./HeroCarousel"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: "Example"
}; // eslint-disable-next-line react/prop-types

exports.default = _default;

const FgSlide = ({
  seed
}) => /*#__PURE__*/_react.default.createElement("img", {
  style: {
    width: "100%",
    height: "100%",
    objectFit: "cover"
  },
  src: `https://placem.at/places?w=1920&h=1080&random=${seed}&txt=${seed}`,
  alt: `Placeholder ${seed}`
}); // eslint-disable-next-line react/prop-types


const BgSlide = ({
  seed
}) => /*#__PURE__*/_react.default.createElement("article", {
  style: {
    backgroundImage: `url(https://placem.at/places?w=1920&h=1080&random=${seed}&txt=${seed})`,
    backgroundPosition: "center",
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    position: "absolute",
    top: 0,
    right: 0,
    bottom: 0,
    left: 0
  }
});

const Images = () => /*#__PURE__*/_react.default.createElement(_HeroCarousel.default, null, /*#__PURE__*/_react.default.createElement(FgSlide, {
  seed: "1"
}), /*#__PURE__*/_react.default.createElement(FgSlide, {
  seed: "2"
}), /*#__PURE__*/_react.default.createElement(FgSlide, {
  seed: "3"
}), /*#__PURE__*/_react.default.createElement(FgSlide, {
  seed: "4"
}), /*#__PURE__*/_react.default.createElement(FgSlide, {
  seed: "5"
}));

exports.Images = Images;

const BackgroundImages = () => /*#__PURE__*/_react.default.createElement(_HeroCarousel.default, null, /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "1"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "2"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "3"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "4"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "5"
}));

exports.BackgroundImages = BackgroundImages;

const CustomInterval = () => /*#__PURE__*/_react.default.createElement(_HeroCarousel.default, {
  interval: 8000
}, /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "1"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "2"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "3"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "4"
}), /*#__PURE__*/_react.default.createElement(BgSlide, {
  seed: "5"
}));

exports.CustomInterval = CustomInterval;
